client/components/autocomplete2/list.jsx

```diff
@@ -0,0 +1,87 @@
+import { clone } from 'lodash'
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import { Autocomplete } from '/client/components/autocomplete2/index'
+
+export class AutocompleteList extends Component {
+  static propTypes = {
+    className: PropTypes.string,
+    disabled: PropTypes.bool,
+    filter: PropTypes.func,
+    fetchItems: PropTypes.func,
+    formatSelected: PropTypes.func,
+    formatSearchResult: PropTypes.func,
+    items: PropTypes.array,
+    onSelect: PropTypes.func,
+    placeholder: PropTypes.string,
+    url: PropTypes.string
+  }
+
+  state = {
+    items: []
+  }
+
+  componentWillMount = () => {
+    this.fetchItems()
+  }
+
+  componentDidUpdate = (prevProps) => {
+    if (prevProps.items !== this.props.items) {
+      this.fetchItems()
+    }
+  }
+
+  fetchItems = () => {
+    const { fetchItems } = this.props
+    const { items } = this.state
+
+    fetchItems(items, (fetchedItems) => {
+      this.setState({ items: fetchedItems })
+    })
+  }
+
+  onRemoveItem = (item) => {
+    const { items, onSelect } = this.props
+    const newItems = clone(items)
+
+    newItems.splice(item, 1)
+    onSelect(newItems)
+    this.setState({items: newItems})
+  }
+
+  render () {
+    const { className, formatSelected } = this.props
+    const { items } = this.state
+
+    return (
+      <div className={`Autocomplete--list ${className || ''}`}>
+
+        <div className='Autocomplete__list'>
```

Comment:
Ah took a minute to understand the BEM logic in this className and the one above and not sure if I still do 😛 . Wondering if this could benefit from changing to `AutocompleteList` as the parent and `AutocompleteList__list` or `autocomplete-list` and `autocomplete-list__list` if you prefer dash style spacing?

Reply:
Good call, will change to AutocompleteList -- was using the bem style of variations because the Autocomplete classes are shared between different components, but your suggestion of using component name is clearer.